### PR TITLE
Bring back the cursed thing

### LINKED ISSUE #482: Using internal or any optifine shaders breaks structurize building preview render
### Prerequisites 

- [ ] I am running the latest **alpha** version of Structurize for my Minecraft version.
- [x] I am running the latest version of OptiFine for my Minecraft version.
- [ ] I am running the latest version of this shader for my Minecraft and OptiFine version.
- [ ] I made sure that **this OptiFine issue is not a duplicate of any existing OptiFine issue.**

### Context 

- Minecraft Version: 1.18.1
- Structurize Version: 1.0.326-ALPHA
- OptiFine Version: HD U H4
- Shaders and their Versions:  BSL or internal or any other shaders

## Steps to reproduce the issue

1. Create any world
2. Pick some structure block from Minecolonies (e.g. supply ship)
3. Place it
4. Press Esc and look around

### Logs

- latest.log: https://gist.github.com/Z4X4R0FF/701e82175cc928ff32d796ea20851f40)
- crashlog: 

### Notes
Also i spotted almost equal bug when i turn shaders off completely, leaving only optifne. When placing supply ship with shaders turned off, the water blocks preview teleports away and stands somewhere above the ship preview. 
Also i ve tried to tick on/off some optifine options but this didn't solve issue.
The wrongly placed preview blocks start following player if you press ESC from structurize preview

---
## Viewers

* Add a thumbs-up to this OptiFine issue if you are also affected. This helps the issue become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Nightenom**
Closes #482 

# Changes proposed in this pull request:
- clean up gradle
- add download and setup methods for optifine runtime (not source decompile)
- cleanup code deprecations/few warnings
- add fog to preview rendering (boxes are still visible in fog)
- enable and fix optifine compat (tested on luma and seus shaders, without too)

Review please

### CODE REVIEW

File: gradle/optifine.gradle [hunk 1]
```diff
@@ -1,9 +1,123 @@
...
+                exclude filesToKeep
+            }
+        }
+    }
+    else {
+        if (optifineFileName != null) delete modsFolder.resolve(optifineFileName)
+        if (optifineTweakerFileName != null) delete modsFolder.resolve(optifineTweakerFileName)
+    }
+
+    task cleanOptifine() {
```
Comment (marchermans) [inline gradle/optifine.gradle:118]: Nice work, one minor thing: This should depend on the clean task.

File: src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java [hunk 2]
```diff
@@ -104,6 +117,20 @@ private void setupReflectedMethodReferences() throws ClassNotFoundException, NoS
...
+        final Class<MultiBufferSource> multiBufferSourceClass = MultiBufferSource.class;
+
+        isFogOffMethod = configClass.getMethod("isFogOff");
+        isFogOffMethod.setAccessible(true); 
+
+        fogStandardField = fogRendererClass.getField("fogStandard");
+        fogStandardField.setAccessible(true);
+        
+        setFogAllowedMethod = renderSystemClass.getMethod("setFogAllowed", boolean.class);
+        setFogAllowedMethod.setAccessible(true);
```
Comment (marchermans) [inline src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java:133]: This won't work in production, and you really should not need it either.....

**Review by marchermans — CHANGES_REQUESTED** [review]
So the general concept is good, but I missing a major explanation why you are reflecting into MC code, especially without considering obfuscation....

File: src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java:133]: These are optifine methods/fields in vanilla classes, why should it not work?

File: gradle/optifine.gradle (hunk 1 above)
Comment (Nightenom) [reply gradle/optifine.gradle:118]: hmm - probably clean task should depend on this? so if you run clean task it also run this task
(or clean.finalizedBy(cleanOptifine))

Comment (marchermans) [reply gradle/optifine.gradle:118]: Eh yeah that is what i mean

File: src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java (hunk 2 above)
Comment (marchermans) [reply src/main/java/com/ldtteam/structurize/optifine/OptifineCompat.java:133]: Ah
Optifine adds them...........Oef....

File: gradle/optifine.gradle (hunk 1 above)
Comment (Nightenom) [reply gradle/optifine.gradle:118]: Used `tasks.clean.finalizedBy cleanOptifine`

**Review by marchermans — APPROVED** [review]

**Pull request merged**